CourseProject/src/main/java/com/neoflex/CourseProject/service/ValidationService.java

```diff
@@ -0,0 +1,103 @@
+        boolean resultsOfValidations=true;
+       Boolean[] testResults;
+       if(loanStatementRequestDto.getMiddleName() != null){
+           int maxCountConditions = 9;
+           testResults = new Boolean[maxCountConditions];
+           testResults[8] = isNamesValid(loanStatementRequestDto.getMiddleName());
+       }else{
+           int minCountConditions = 8;
+           testResults = new Boolean[minCountConditions];
+       }
+       testResults[0] = isEmailValid(loanStatementRequestDto.getEmail());
+       testResults[1] = isNamesValid(loanStatementRequestDto.getFirstName());
+       testResults[2] = isNamesValid(loanStatementRequestDto.getLastName());
+       testResults[3] = isPassportSeriesValid(loanStatementRequestDto.getPassportSeries());
+       testResults[4] = isPassportNumberValid(loanStatementRequestDto.getPassportNumber());
+       testResults[5] = isAmountMeetsTheCondition(loanStatementRequestDto.getAmount());
+       testResults[6] = isTermMeetsTheCondition(loanStatementRequestDto.getTerm());
+       testResults[7] = isAdult(loanStatementRequestDto.getBirthdate());
+       return resultsOfValidations;
+   }
+
+    Boolean isEmailValid(String email) throws ValidationException {
+       String mailRegex = "^[A-Za-z0-9+_.-]+@(.+)$";
+       Pattern pattern = Pattern.compile(mailRegex);
+       Matcher matcher = pattern.matcher(email);
+       if(matcher.matches())
+       return true;
+       else
+           throw new ValidationException("email is not valid " + email);
+   }
+
+     Boolean isNamesValid(String name) throws ValidationException {
+        String nameRegex = "^[A-Za-z]{2,30}";
+        Pattern pattern = Pattern.compile(nameRegex);
+        Matcher matcher = pattern.matcher(name);
+         if(matcher.matches())
+             return true;
+         else
+             throw new ValidationException("name is not valid " + name);
+    }
+
+     Boolean isPassportSeriesValid(String passportSeries) throws ValidationException {
+        String passportSeriesRegex="^[0-9]{4}";
+        Pattern pattern = Pattern.compile(passportSeriesRegex);
+        Matcher matcher = pattern.matcher(passportSeries);
+         if(matcher.matches())
+             return true;
+         else
+             throw new ValidationException("passport series is not valid " + passportSeries);
+    }
+
+     Boolean isPassportNumberValid(String passportNumber) throws ValidationException {
+        String passportSeriesRegex="^[0-9]{4}";
+        Pattern pattern = Pattern.compile(passportSeriesRegex);
+        Matcher matcher = pattern.matcher(passportNumber);
+         if(matcher.matches())
+             return true;
+         else
+             throw new ValidationException("passport number is not valid " + passportNumber);
+    }
+
+     Boolean isAmountMeetsTheCondition(BigDecimal amount) throws ValidationException {
+       if(amount.compareTo(BigDecimal.valueOf(20000)) >= 0)
```

Comment:
Во всех конструкциях убери проверку на true, нам хватит проверки на то, что данные не удовлетворяют всем кондициям, и прокинем эксепшены. Избавишься в таком случае от лишних else'ов.
```suggestion
       if(!amount.compareTo(BigDecimal.valueOf(20000)) >= 0) {
           throw new ValidationException("amount is too small " + amount);
       }
```

Reply:
Но в данном случае гораздо нагляднее будет выглядеть таким образом: 
```suggestion
       if(amount.compareTo(BigDecimal.valueOf(20000)) < 0) {
           throw new ValidationException("amount is too small " + amount);
       }
```